This patch is based on the upstream commit described below, adapted for use
in the Debian package by Peter Michael Green.

commit edd0f60d0baf09604553525c2636df5d6ba05d44
Author: Ian Douglas Scott <idscott@system76.com>
Date:   Fri Sep 22 17:47:30 2023 -0700

    Use `rustix` instead of `nix`

Index: wayland-server/Cargo.toml
===================================================================
--- wayland-server.orig/Cargo.toml
+++ wayland-server/Cargo.toml
@@ -50,9 +50,9 @@ version = "2.0.0"
 version = "0.4"
 optional = true
 
-[dependencies.nix]
-version = "0.26.0"
-default-features = false
+[dependencies.rustix]
+version = "0.38.14"
+features = ["fs"]
 
 [dependencies.wayland-backend]
 version = "0.1.0"
Index: wayland-server/src/socket.rs
===================================================================
--- wayland-server.orig/src/socket.rs
+++ wayland-server/src/socket.rs
@@ -15,7 +15,7 @@ use std::{
 };
 
 use io_lifetimes::{AsFd, BorrowedFd};
-use nix::fcntl::{flock, FlockArg};
+use rustix::fs::{flock, FlockOperation};
 
 /// An utility representing a unix socket on which your compositor is listening for new clients
 #[derive(Debug)]
@@ -91,7 +91,7 @@ impl ListeningSocket {
                 .map_err(|_| BindError::PermissionDenied)?;
 
             // lock the lockfile
-            if flock(_lock.as_raw_fd(), FlockArg::LockExclusiveNonblock).is_err() {
+            if flock(&_lock, FlockOperation::NonBlockingLockExclusive).is_err() {
                 return Err(BindError::AlreadyInUse);
             }
 
